requests_cache/backends/base.py

```diff
@@ -170,12 +172,39 @@ def restore_response(self, response, seen=None):
         result.history = tuple(self.restore_response(r, seen) for r in response.history)
         return result
 
+    def _remove_ignored_parameters(self, request):
+
+        def filter_ignored_parameters(d):
+            return [(k, v) for k, v in sorted(d.items()) if k not in self._ignored_parameters]
+
+        url = urlparse(request.url)
+        query = parse_qs(url.query)
+        query = filter_ignored_parameters(query)
+        query = urlencode(query)
+        url = urlunparse((url.scheme, url.netloc, url.path, url.params, query, url.fragment))
+        body = request.body
+        if body:
+            if request.headers['content-type'] == 'application/json':
```

Comment:
You can pass raw data without headers, e.g.
```python
r = s.post(url, data="raw data")
```
1. ``request.headers['content-type']`` will raise ``KeyError``. ``headers.get(...)`` will solve it
2. ``parse_qs(request.body)`` will return empty dict, so body will be wrong. I think there must be check 
``if request.body and not body: use request.body, else: filter, encode``

Also, please write test for this case.